google/cloud/testing_util/validate_metadata.cc

```diff
@@ -89,7 +89,13 @@ RoutingHeaders ExtractMDFromHeader(std::string header) {
  */
 MATCHER_P(MatchesGlob, glob, "matches the glob: \"" + glob + "\"") {
   // Translate the glob into a regex pattern.
-  std::regex regex(absl::StrReplaceAll(glob, {{"*", "[^/]+"}}));
+  auto matcher1 = absl::StrReplaceAll(glob, {{"*", "[^/]+"}});
+  // Create a second matcher that replaces all "/" with "%2F" to match the
+  // character replacement in `internal::UrlEncode`.
+  auto decoded_glob = absl::StrReplaceAll(glob, {{"/", "%2F"}});
+  auto matcher2 = absl::StrReplaceAll(decoded_glob, {{"*", "[^/]+"}});
+  auto regex_string = absl::StrCat(matcher1, "|", matcher2);
```

Comment:
So i think the reason we are checking for either is because of what the test is passing in.

Some tests pass in` projects/*/resource/*` while others are passing in `projects%2F*%2Fresource%2F*%2F`

It seems easier to understand/reason (as a human) if when we look at these tests we can see the / instead of the escaped characters.

Reply:
> Some tests pass in projects/*/resource/* while others are passing in projects%2F*%2Fresource%2F*%2F

nit: Some tests pass in `projects/*/resource/*` while others are passing in `projects%2Ffoo%2Fresource%2Fbar%2F`. I think the implicit ones (with `*`s) do not encode, and the explicit ones (without `*`s) do.

We should find a way to verify that URL encoding works without false positives.

> It seems easier to understand/reason (as a human) if when we look at these tests we can see the / instead of the escaped characters.

Yeah, I think I agree. That can be made to work by saying `UrlEncode("foo/bar/baz")` in the tests.